api/collections/serializers.py

```diff
@@ -1,13 +1,35 @@
 from django.db import IntegrityError
+from rest_framework import exceptions
 from rest_framework import serializers as ser
 
-from osf.models import AbstractNode, Node, Collection, Registration
+from osf.models import AbstractNode, Node, Collection, CollectedGuidMetadata, Guid, Registration, AbstractProvider
 from osf.exceptions import ValidationError
 from api.base.serializers import LinksField, RelationshipField, LinkedNodesRelationshipSerializer, LinkedRegistrationsRelationshipSerializer
 from api.base.serializers import JSONAPISerializer, IDField, TypeField, VersionedDateTimeField
 from api.base.exceptions import InvalidModelValueError, RelationshipPostMakesNoChanges
 from api.base.utils import absolute_reverse, get_user_auth
 from api.nodes.serializers import NodeLinksSerializer
+from api.taxonomies.serializers import TaxonomizableSerializerMixin
+from framework.exceptions import PermissionsError
+from osf.utils.permissions import WRITE
+from website.exceptions import NodeStateError
+
+
+class ProviderRelationshipField(RelationshipField):
+    def get_object(self, provider_id):
+        return AbstractProvider.load(provider_id)
+
+    def to_internal_value(self, data):
+        provider = self.get_object(data)
+        return {'provider': provider}
+
+class GuidRelationshipField(RelationshipField):
+    def get_object(self, _id):
+        return Guid.load(_id)
+
+    def to_internal_value(self, data):
+        guid = self.get_object(data)
+        return {'guid': guid}
```

Comment:
Is there advantage to doing it this way versus loading the provider and `Guid` in the `.create` method?

Reply:
Thanks for the clarification. Might want to add a code comment that this is necessary because `Serializer.run_validation` expects `to_internal_value` to return a `dict`.

Reply:
We're doing the same thing for other models in a number of other places; might also be worth filing a ticket to make a generic version of this field that just accepts an `internal_value_key` and `model_class` on `__init__` to do this automagically.